dallinger/experiment.py

```diff
@@ -408,6 +421,25 @@ def attention_check(self, participant):
         """
         return True
 
+    def participant_task_completed(self, participant):
+        """Called when an experiment task is finished, generally from the
+        /worker_complete route.
+
+        Assign the qualifications to the Participant, via the recruiter.
+        These will include one Qualification for the experiment
+        ID, and others for the configured group_name, if it's been set.
+
+        Overrecruited participants don't receive qualifications, since they
+        haven't actually completed the experiment. This allows them to remain
+        eligible for future runs. Blah blah blah.. just an example.
+        """
+        if participant.status == "overrecruited":
```

Comment:
Is "overrecruited" a base status? I didn't know it existed which makes me a little queasy to see other code relying on it. I assume this is a way to not give participants the qualification if they didn't meaningfully take part. I wonder if there is a way to do this without relying on the status code, which users might not use correctly (I wouldn't have in this case).

Reply:
It is. See https://github.com/Dallinger/Dallinger/blob/master/dallinger/models.py#L238 It was introduced once Jordan and others found significant overrecruitment to be an effective strategy for getting a quorum in large waiting room based experiments.

Reply:
I just noticed it's missing from the comment block that generates the Sphinx documentation. I don't know if this was intentional. :-/

Reply:
Got it. I guess it's kinda my fault. I've never used the built in waiting room and just make my own, so I've never bumped into any of this code. It would mean that this check would fail for me though, but that at least wouldn't break my intuition for qualification handling, I would just miss out on a few things.